ionoscloud/resource_server.go

```diff
@@ -1001,15 +1015,15 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 			mProp, _ := json.Marshal(nicProperties)
 			log.Printf("[DEBUG] Updating props: %s", string(mProp))
 			var createdNic ionoscloud.Nic
-			if !updateNic {
+			if createNic {
 				nic.Properties = &nicProperties
 				createdNic, apiResponse, err = client.NetworkInterfacesApi.DatacentersServersNicsPost(ctx, d.Get("datacenter_id").(string), *server.Id).Nic(*nic).Execute()
-			} else {
+			} else if nic.Id != nil {
 				_, apiResponse, err = client.NetworkInterfacesApi.DatacentersServersNicsPatch(ctx, d.Get("datacenter_id").(string), *server.Id, *nic.Id).Nic(nicProperties).Execute()
 			}
 			logApiRequestTime(apiResponse)
 			if err != nil {
-				diags := diag.FromErr(fmt.Errorf("error updating nic (%s)", err))
+				diags := diag.FromErr(fmt.Errorf("error nic (%w)", err))
```

Comment:
@cristiGuranIonos why did we remove `updating` from here? `error updating nic: %w` makes sense to me

Reply:
it's not clear if it's an update or a create that errored.